lenskit-core/src/main/java/org/lenskit/rerank/GreedyRerankingItemRecommender.java

```diff
@@ -70,19 +72,16 @@ protected ResultList recommendWithDetails(long user, int n, @Nullable LongSet ca
 
         List<Result> results = new ArrayList<>(n);
         for (int i = 0; i<n; i++) {
-            Result nextItem = selector.nextItem(user, n, results, candidates);
+            final Result nextItem = strategy.nextItem(user, n, results, candidates);
             if (nextItem == null) {
                 break;
             } else {
-                //TODO: REVIEW: there should be an easier way to do this.
-                // Remove the selected item from the list of candidate items.
-                for(Iterator<Result> it = candidates.iterator(); it.hasNext();) {
-                    Result listResult = it.next();
-                    if (listResult.getId() == nextItem.getId()) {
-                        it.remove();
-                        break;
+                Iterables.removeIf(candidates, new Predicate<Result>() {
```

Comment:
We can simplify this by adding an `idEquivalence` to `Results`, and then this becomes:

```java
Iterables.removeIf(candidates, Results.idEquivalence().equalTo(nextItem));
```

However, we can do that as a separate cleanup pass.